app/models/solidus_square/gateway.rb

```diff
@@ -55,6 +56,10 @@ def create_customer(user, address)
       ::SolidusSquare::Customers::Create.call(client: client, spree_user: user, spree_address: address)
     end
 
+    def purchase(amount, payment_source, gateway_options)
+      authorize(amount, payment_source, gateway_options)
```

Comment:
Just a minor thing.
It sounds weird to call a method `authorize` that capture the payment 🤔 

What do you think to create a private method `create_payment_on_square` to do:

```
def authorize(amount, payment_source, gateway_options)
  create_square_payment(amount, payment_source, gateway_options)
end

def purchase(amount, payment_source, gateway_options)
  create_square_payment(amount, payment_source, gateway_options)
end

private

def create_square_payment(amount, payment_source, gateway_options)
     payment = gateway_options[:originator]
      source_id = payment_source.nonce
      auto_capture = payment.payment_method.auto_capture
      square_payment = create_payment(amount, source_id, auto_capture)
      payment.response_code = square_payment[:id]
      payment_source.update!(payment_source_constructor(square_payment))

      ActiveMerchant::Billing::Response.new(true, 'Transaction approved', square_payment,
        authorization: square_payment[:id])
    rescue StandardError => e
      ActiveMerchant::Billing::Response.new(false, e.message, {})
end
```

Reply:
I 100% agree. I don't know why I didn't think about that!

Reply:
https://github.com/nebulab/solidus_square/pull/63/commits/3e61e310f9c3e4f63d9a80536130e6df5086ce44